code/part-one/tests.js

```diff
@@ -0,0 +1,129 @@
+const chai = require('chai');
+const expect = chai.expect;
+const { hash } = require('./utils/helpers.js');
+const { Block, Blockchain, Transaction } = require('./blockchain.js');
+
+
+// Test transaction data
+const testFromAddress = hash('testFromAddress');
+const testToAddress = hash('testToAddress');
+const testAmount = '10.00';
+const transaction = new Transaction(
+  testFromAddress,
+  testToAddress,
+  testAmount
+);
+
+// Test blockchain data
+const blockchain = new Blockchain();
+const genesisBlock = blockchain.chain[0];
+const newBlock = new Block();
+const miningAddress = hash('miningAddress');
+let timesMined = 0;
+
+// Test block data
+const testHash = '0123TESTHASH';
+const testTransactions = [];
+const testNonce = 0;
+const now = Date.now();
+const block = new Block(now, testTransactions, testHash);
+const testDataHash = hash(
+  testHash + now + JSON.stringify(testTransactions) + testNonce
+);
+
+
+// Transaction tests
+describe('Transaction', () => {
+  it('Should create a transaction with all initial values', done => {
+    expect(transaction.fromAddress).to.equal(testFromAddress);
+    expect(transaction.toAddress).to.equal(testToAddress);
+    expect(transaction.amount).to.equal(testAmount);
+    done();
+  });
+});
+
+// Blockchain tests
+describe('Blockchain', () => {
+  it('Should create a blockchain with a genesis block', done => {
+    expect(blockchain).to.have.property('_createGenesisBlock');
+    expect(blockchain).to.have.property('chain').with.lengthOf(1);
```

Comment:
@therobinkim Do you have an opinion on whether or not we include `expect(...).to.have.property(...)` tests? It seems to me that the API should probably be included in the stub file the students get, and if we do test that the API hasn't changed, it's in one initial test that passes before the students modify anything.

Reply:
@delventhalz I have no strong preference of testing existence of properties